src/main/java/seedu/address/model/supplyItem/SupplyItem.java

```diff
@@ -0,0 +1,114 @@
+package seedu.address.model.supplyItem;
+
+import static seedu.address.commons.util.CollectionUtil.requireAllNonNull;
+
+import java.util.HashSet;
+import java.util.Objects;
+import java.util.Set;
+
+import seedu.address.model.person.Name;
+import seedu.address.model.person.Person;
+import seedu.address.model.person.Price;
+import seedu.address.model.tag.Tag;
+
+/**
+ * Represents a Supply Item in Salesy
+ * Guarantees: details are present and not null, field values are validated, immutable.
```

Comment:
Does this imply that attributes are not supposed to be changed?

Reply:
Similar to `Person` class, just guarantees that all fields are `private final` i.e. immutable. 

```
SupplyItem item = new SupplyItem(...);
item.currentStock = 9999; // this will fail due to immutable property of `private final`
```

Reply:
Can still edit the item via creation of a new class similar to `Persons`

Reply:
Understood thanks